pkg/restapi/device/operations.go

```diff
@@ -208,6 +258,51 @@ func (o *Operation) finishRegistration(w http.ResponseWriter, r *http.Request) {
 	logger.Debugf("Registration success")
 }
 
+func (o *Operation) requestDeviceValidation(userSub, aaguid string, certs []interface{}) error {
+	if len(certs) == 0 {
+		return fmt.Errorf("missing certs")
+	}
+
+	var certPemList []string
+
+	for _, certInterface := range certs {
+		cert, ok := certInterface.([]byte)
+		if !ok {
+			return fmt.Errorf("can't cast certificate data to []byte")
+		}
+
+		certPemList = append(certPemList, string(pem.EncodeToMemory(
+			&pem.Block{Bytes: cert, Type: "CERTIFICATE"},
+		)))
+	}
+
+	postData, err := json.Marshal(&struct {
+		X5c    []string `json:"x5c"`
+		Sub    string   `json:"sub"`
+		Aaguid string   `json:"aaguid"`
+	}{
+		X5c:    certPemList,
+		Sub:    userSub,
+		Aaguid: aaguid,
+	})
+	if err != nil {
+		return fmt.Errorf("failed to marshal cert data: %w", err)
+	}
+
+	req, err := http.NewRequestWithContext(context.TODO(),
```

Comment:
note: you should use the HTTP request's context

Reply:
done